process/headerCheck/headerSignatureVerify.go

```diff
@@ -154,27 +153,32 @@ func (hsv *HeaderSigVerifier) getConsensusSigners(header data.HeaderHandler) ([]
 
 // VerifySignature will check if signature is correct
 func (hsv *HeaderSigVerifier) VerifySignature(header data.HeaderHandler) error {
-	multiSigVerifier, err := hsv.multiSigContainer.GetMultiSigner(header.GetEpoch())
+	headerCopy, err := hsv.copyHeaderWithoutSig(header)
 	if err != nil {
 		return err
 	}
 
-	headerCopy, err := hsv.copyHeaderWithoutSig(header)
+	hash, err := core.CalculateHash(hsv.marshalizer, hsv.hasher, headerCopy)
 	if err != nil {
 		return err
 	}
 
-	hash, err := core.CalculateHash(hsv.marshalizer, hsv.hasher, headerCopy)
+	return hsv.VerifySignatureForHash(header, hash, header.GetPubKeysBitmap(), header.GetSignature())
+}
+
+// VerifySignatureForHash will check if signature is correct for the provided hash
+func (hsv *HeaderSigVerifier) VerifySignatureForHash(header data.HeaderHandler, hash []byte, pubkeysBitmap []byte, signature []byte) error {
+	multiSigVerifier, err := hsv.multiSigContainer.GetMultiSigner(header.GetEpoch())
```

Comment:
so, supposing that we just started a new epoch. When we call this function on the previous header (that was on the previous epoch) we verify the signature generated in the new epoch but with the previous epoch consensus group.